app/controllers/Controller.php

```diff
@@ -1,27 +1,136 @@
 <?php
+
 namespace Application\Controllers;
 
+use Application\Core\Auth\Auth;
 use Twig\Environment;
 use Twig\Loader\FilesystemLoader;
 
 /**
  * make working twig by heritage
  */
-abstract class Controller
+abstract class Controller extends Auth
 {
-        private $Loader;
-        protected $twig;
+    protected $twig;
+
+    /**
+     * main controller
+     * integration twig
+     * 
+     */
+    public function __construct()
+    {
+        // where the twig views
+        $this->loader = new FilesystemLoader(ROOT . '/app/Views');
+        // env twig
+        // put true for debug // prod
+        $this->twig = new Environment(
+            $this->loader,
+            [
+                'debug' => false
+            ]
+        );
+        $this->twig->addGlobal('session', $_SESSION);
+    }
+
+    /**
+     * function to make message readeableby twig
+     *
+     * @param string $message
+     * @return Array
+     */
+    public function setMessageForTwig(string $message)
+    {
+        $messageReadle = $message;
+        $arrayMessage = array(
+            "message" => $messageReadle
+        );
+        return $arrayMessage;
+    }
+
+    /**
+     * function to void session
+     *
+     * @return void
+     */
+    public function InitSession()
+    {
+        $_SESSION = array();
+        unset($_SESSION['LOGGED_USER']);
+        unset($_SESSION['LOGGED_USER_NAME']);
+        unset($_SESSION['LOGGED_USER_ID']);
+        unset($_SESSION);
+        session_destroy();
+    }
 
-        public function __construct()
-        {
-                $this->loader = new FilesystemLoader(ROOT . '/app/templates');
+    /**
+     *  function to verify that the user is logged
+     *
+     * @return void
+     */
+    public function verifyComment($message)
+    {
+        if (($_SESSION['LOGGED_USER_NAME'] == "")) {
```

Comment:
```suggestion
        if ($_SESSION['LOGGED_USER_NAME'] != "") return
```
Tu peux faire des early return, c'est une bonne pratique de code, en générale pratiqué par des dev avec de XP. Mais ça reste du code style

Reply:
@castroCrea 

So it's good or a have to change ?